File: InclusiveConnectBackend/src/main/java/com/inclusiveconnect/inclusiveconnectbackend/service/ConnectionService.java
package com.inclusiveconnect.inclusiveconnectbackend.service;

import com.inclusiveconnect.inclusiveconnectbackend.dto.response.ConnectionRequestResponse;
import com.inclusiveconnect.inclusiveconnectbackend.dto.response.ConnectionResponse;

import java.util.List;

public interface ConnectionService {
    ConnectionRequestResponse sendRequest(Long senderId, Long receiverId);

    ConnectionRequestResponse acceptRequest(Long receiverId, Long requestId);

    ConnectionRequestResponse rejectRequest(Long receiverId, Long requestId);

    List<ConnectionRequestResponse> getPendingRequests(Long userId);

    List<ConnectionRequestResponse> getSentRequests(Long userId);

    List<ConnectionResponse> getMyConnections(Long userId);

    void removeConnection(Long currentUserId, Long targetUserId);

    void cancelRequest(Long senderId, Long requestId);
}

File: InclusiveConnectFrontend/src/app/core/services/connection.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models/api-response.model';

export interface Connection {
    userId: number;
    name: string;
    headline?: string;
    profilePicture?: string;
}

export interface ConnectionRequest {
    id: number;
    status: string;
    sentAt: string;
    senderId: number;
    senderName: string;
    senderHeadline?: string;
    receiverId: number;
    receiverName: string;
    receiverHeadline?: string;
}

@Injectable({ providedIn: 'root' })
export class ConnectionService {
    private http = inject(HttpClient);
    private baseUrl = `${environment.apiUrl}/connections`;

    getMyConnections(): Observable<ApiResponse<Connection[]>> {
        return this.http.get<ApiResponse<Connection[]>>(this.baseUrl);
    }

    getPendingRequests(): Observable<ApiResponse<ConnectionRequest[]>> {
        return this.http.get<ApiResponse<ConnectionRequest[]>>(`${this.baseUrl}/pending`);
    }

    sendRequest(userId: number): Observable<ApiResponse<ConnectionRequest>> {
        return this.http.post<ApiResponse<ConnectionRequest>>(`${this.baseUrl}/request/${userId}`, {});
    }

    acceptRequest(requestId: number): Observable<ApiResponse<ConnectionRequest>> {
        return this.http.put<ApiResponse<ConnectionRequest>>(`${this.baseUrl}/accept/${requestId}`, {});
    }

    rejectRequest(requestId: number): Observable<ApiResponse<ConnectionRequest>> {
        return this.http.put<ApiResponse<ConnectionRequest>>(`${this.baseUrl}/reject/${requestId}`, {});
    }

    getSentRequests(): Observable<ApiResponse<ConnectionRequest[]>> {
        return this.http.get<ApiResponse<ConnectionRequest[]>>(`${this.baseUrl}/sent`);
    }

    removeConnection(userId: number): Observable<ApiResponse<void>> {
        return this.http.delete<ApiResponse<void>>(`${this.baseUrl}/${userId}`);
    }

    cancelRequest(requestId: number): Observable<ApiResponse<void>> {
        return this.http.delete<ApiResponse<void>>(`${this.baseUrl}/request/${requestId}`);
    }
}
